.github/workflows/build.yml

```diff
@@ -0,0 +1,81 @@
+name: build
+on:
+  push:
+#    branches: [ master ] Todo: Uncomment before merging
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release? yes/no'
+        default: 'no'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: eu-west-1
+      - name: Compile
```

Comment:
I used [the official Gradle action](https://github.com/gradle/gradle-build-action) here and the reason for that is that it handles caches for us so the whole pipeline configuration is cleaner and shorter (no need for a separate `cache gradle` step as it was in the CircleCi config).

Reply:
That's nice, less boilerplate FTW